### PR TITLE
Improve scalability of various buttons.

### PR DESCRIPTION
**Pull request by Unengine**
### Description

SSIA

### How to test

Check if following prefabs are working properly in game.

MailCell
QuestCell
ArenaRankCell
UI_CodeRewardPopup
UI_HelpTooltip
UI_ItemCountAndPricePopup
UI_ItemCountableAndPricePopup
UI_ItemInformationTooltip
UI_LoginSystem
UI_MonsterCollectionRewardsPopup
UI_StageInformation
UI_TwoButtonSystem
UI_ConfirmPopup
UI_OneButtonSystem
UI_RankingBattleResultPopup
UI_Craft
UI_UpgradeEquipment
UI_QuestPreparation
UI_MimisbrunnrPreparation
UI_CombinationSlotPopup

### Related Links

https://app.asana.com/0/958521740385861/1201268185760776
https://app.asana.com/0/1133453747809944/1201249334676635
https://app.asana.com/0/1133453747809944/1201425663841592
https://app.asana.com/0/1133453747809944/1201430475653447
https://app.asana.com/0/1133453747809944/1201425663841568
https://app.asana.com/0/1133453747809944/1201407631847062
https://app.asana.com/0/1133453747809944/1201425663841592
https://app.asana.com/0/1133453747809944/1201425663841600
https://app.asana.com/0/1133453747809944/1201430475653457

### Required Reviewers

@planetarium/9c-dev @Namyujeong (optional)

### CODE REVIEW

**Comment by hskim881028** [comment]
This PR looks good to be reviewed by call

**Comment by sonohoshi** [comment]
image
It's OK?

File: nekoyume/Assets/_Scripts/UI/Module/TextButton.cs [hunk 1]
```diff
@@ -1,83 +1,34 @@
...
-        public Image backgroundImage;
-        public Text text;
-        
-        private Model.TextButton _data;
+        [SerializeField]
+        private TextMeshProUGUI text = null;
         
-        private readonly List<IDisposable> _disposablesForAwake = new List<IDisposable>();
-        private readonly List<IDisposable> _disposablesForSetData = new List<IDisposable>();
+        [SerializeField]
```
Comment (boscohyun) [inline nekoyume/Assets/_Scripts/UI/Module/TextButton.cs:13]: It seems can remove `SerializeField` attribute from the `button` if the `button` assigned in `Awake()` with `RequiredComponent(typeof(Button))` attribute.

File: nekoyume/Assets/_Scripts/UI/Widget/Popup/ChatPopup.cs [hunk 2]
```diff
@@ -1,4 +1,4 @@
-﻿
+
```
Comment (boscohyun) [inline nekoyume/Assets/_Scripts/UI/Widget/Popup/ChatPopup.cs:1]:
```suggestion
```

File: nekoyume/Assets/_Scripts/UI/Widget/Popup/ChatPopup.cs [hunk 3]
```diff
@@ -18,7 +18,6 @@ public override void Initialize()
             {
                 Application.OpenURL(GameConfig.DiscordLink);
                 HelpTooltip.HelpMe(100012, true);
-                AudioController.PlayClick();
```
Comment (boscohyun) [inline nekoyume/Assets/_Scripts/UI/Widget/Popup/ChatPopup.cs:21]: Is this intended?

File: nekoyume/Assets/_Scripts/UI/Widget/Popup/ItemCountPopup.cs [hunk 4]
```diff
@@ -36,37 +34,31 @@ protected override void Awake()
...
-                    {
-                        _data?.OnClickCancel.OnNext(_data);
-                        AudioController.PlayCancel();
-                    })
-                    .AddTo(_disposablesForAwake);
-                CloseWidget = cancelButton.onClick.Invoke;
+                cancelButton.OnClick = () =>
+                {
+                    _data?.OnClickCancel.OnNext(_data);
+                    AudioController.PlayCancel();
```
Comment (boscohyun) [inline nekoyume/Assets/_Scripts/UI/Widget/Popup/ItemCountPopup.cs:48]: Why these `AudioContoller.PlayXxx()` invoke used or not used in same `TextButton` component's click method?

**Comment by Unengine** [comment]
> image It's OK?

This is localization labeling issue and i just fixed it.

File: nekoyume/Assets/_Scripts/UI/Widget/Popup/ChatPopup.cs (hunk 3 above)
Comment (Unengine) [reply nekoyume/Assets/_Scripts/UI/Widget/Popup/ChatPopup.cs:21]: I set this event in inspector. So removed several similar parts.

File: nekoyume/Assets/_Scripts/UI/Widget/Popup/ItemCountPopup.cs (hunk 4 above)
Comment (Unengine) [reply nekoyume/Assets/_Scripts/UI/Widget/Popup/ItemCountPopup.cs:48]: I just intended to keep legacy `OnClick` method, but seems like this breaks the consistency. Just moved it to component's `OnClick` event in inspector.

**Review by boscohyun — APPROVED** [review]

**Pull request merged**